service/history/taskProcessor.go

```diff
@@ -304,19 +311,39 @@ func (t *taskProcessor) ackTaskOnce(
 }
 
 func (t *taskProcessor) initializeLoggerForTask(
-	task *persistence.TimerTaskInfo,
+	task queueTaskInfo,
 ) log.Logger {
 
 	logger := t.logger.WithTags(
 		tag.ShardID(t.shard.GetShardID()),
-		tag.WorkflowDomainID(task.DomainID),
-		tag.WorkflowID(task.WorkflowID),
-		tag.WorkflowRunID(task.RunID),
 		tag.TaskID(task.GetTaskID()),
 		tag.FailoverVersion(task.GetVersion()),
-		tag.TaskType(task.GetTaskType()),
-		tag.WorkflowTimeoutType(int64(task.TimeoutType)),
-	)
-	logger.Debug(fmt.Sprintf("Processing timer task: %v, type: %v", task.GetTaskID(), task.GetTaskType()))
+		tag.TaskType(task.GetTaskType()))
+
+	switch task := task.(type) {
+	case *persistence.TimerTaskInfo:
+		logger = logger.WithTags(
+			tag.WorkflowDomainID(task.DomainID),
+			tag.WorkflowID(task.WorkflowID),
+			tag.WorkflowRunID(task.RunID),
+			tag.WorkflowTimeoutType(int64(task.TimeoutType)),
+		)
+		logger.Debug(fmt.Sprintf("Processing timer task: %v, type: %v", task.GetTaskID(), task.GetTaskType()))
```

Comment:
should tag transfer task / replication task just like timer task

one more thing, replication queue may not be necessary since we are migrating out of Kafka, for cross DC

Reply:
yeah. can just be removed then..